base/deprecated.jl

```diff
@@ -112,6 +112,18 @@ macro deprecate_binding(old, new, export_old=true)
          Expr(:call, :deprecate, __module__, Expr(:quote, old)))
 end
 
+macro deprecate_moved(old, new, export_old=true)
+    eold = esc(old)
+    return Expr(:toplevel,
+         :(function $eold(args...; kwargs...)
+               error($eold, " has been moved to the package ", $new, ".jl.\n",
+                     "Run `Pkg.add(\"", $new, "\")` to install it, restart Julia,\n",
```

Comment:
If `new` has already been "loaded" (i.e. you already ran `using new` and it exports the `eold` symbol), then it will resolve `eold` to the `new` module and you won't see any error message.

If you *haven't* yet run `using new` in the current scope, then even if we could do `eval(:(import $new))` and pass through the arguments to `new.eold(...)` with a warning message, I'm skeptical that we should.  I doubt that Base should ever automatically import an external package = as part of a deprecation.